lib/utilities/retrieveData.js

```diff
@@ -0,0 +1,34 @@
+function retrieveData(locations, dataRetrievalFn,
+    response, logger, errorHandlerFn) {
+    if (locations.length === 0) {
```

Comment:
I'm shifting from the array and calling the function recursively so when all is finished the array length is 0.